### ISSUE TITLE
[1.8.9] IC2 API Compatibility?

### ISSUE THREAD

**Issue by oitsjustjose** [BluSunrize/ImmersiveEngineering#1087]
Alright, let me preface this with: **I know you guys have bigger fish to fry. This is already an incredible build of IE, but I'm just poking and prodding**.

Alright. Now that that disclaimer is over, I wanna ask - is IC2 compatibility planned to be re-added? I honestly loved that more than anything, mostly because all of IC2's methods of power generation are very bland or overdone to me. I'm only asking because I was looking and reading everywhere I could and didn't see any mention of it's dismissal (or eventual re-addition) from IE, so I didn't know what the plan was :)

Side note: it's already been kinda reported, but there's been some weird behavior with wires & connectors - are you now supposed to use LV / MV wire relays like you had to use HV relays? I had a connection coming from a water wheel (where I'm certain the dynamo was producing IF) and moving wire about 40 blocks away using LV wire - was that too low of a voltage, or a possible bug I should report separately?

Anyways - good work you guys, enjoying this build of IE, bugs included ;D

**Comment by malte0811** [comment]
I don't know about IC2 compat, if it does not get added back in I might write an addon that adds the compat (using ASM, so quite ugly).
Yes, you have to use relays for all voltages now.

**Comment by BluSunrize** [comment]
I honestly don't know about IC2 yet. All that integration ever did for me was get certain IC2 devs unreasonably angry and it turned IE into "The power converter mod" for FTB Infinity, no thanks to the pack developers.
Basically, two things I never wanted, drama with other tech mods and neglection in favour of other tech mods.

So yeah, currently unsure whether it'll go back in, but I'll keep the option open for myself.

**Comment by malte0811** [comment]
I understand the second argument, but the first one (drama with ic2 devs) is kind of problematic: the dev you are talking about is not a dev of the "official" version of ic2, he does not even have an account on their jenkins server. He is basically a maker of a remake of ic2 and while I do understand that having him reporting "bugs" is quite annoying and nerve-wrecking it kind of feels wrong to remove compatibility with the "official" ic2 version because of an addon and a remake. We could just add the remake and the addon to the unsupported mod list.

**Comment by SynfulChaot** [comment]
Along these lines, I'd love to at still have the ability for EU power transmission over your lines, even without automatically converting.  Actually **especially** without automatically converting.  Could throw in some cool multiblocks akin to real world AC to DC converters/inverters to handle the conversion if you still wanted it.

**Comment by XFactHD** [comment]
Transferring EU over RF (or IF) cables means that the energy has to be converted into RF (or IF).

**Comment by SynfulChaot** [comment]
Would probably be best with their own lines, but that would probably be better in something like Immersive Integration.

**Comment by Aroma1997** [comment]
Quote from BluSunrize: "All that integration ever did for me was get certain IC2 devs unreasonably angry"
You did not make any of us angry.
You may have gotten angry responses from Speiger, but he is not part of the IC2 dev team in any way and should not count as representat for the IC2 dev team. He has nothing to do with IC2 itself and he never did.

Regarding IC2 integration: If you need any help with it or additional API hooks in IC2, please let me know.

**Comment by malte0811** [comment]
Yes,  I said that before. The other part of @BluSunrize's argument was that IC2 integration caused IE to become a "power converter mod" in FTB Infinity. I thought about making an IE addon that would have a seperate set of connectors for IC2 power transfer and an optional (config) RF transformer when this issue was reported, but I never got around to writing any code for it. If @BluSunrize isn't opposed to me putting IC2 compat into an addon, I will start writing it this week. Just as a note, I recently started rewriting an old non-MC project, so I won't be spending as much time modding as I did before.

**Comment by BluSunrize** [comment]
@Aroma1997 I was unaware of his position be it on the team or not, but that's honestly a minor reason (I decided to ignore him, basically)
Infinity's use was a low blow, and I didn't enjoy what IE was reduced to in it. Also, I don't think Player was ever really happy with what I did either, and I can understand him. Honestly, at this point I myself feel more convinced that IC2's energy system should stay separate. It's sufficiently different, unique, and has stood alone throughout modding history, a feat I wouldn't want to undermine. It's what makes IC2 special among the techmods, and I have respect for that.
And I would not want to break your folks' balance by enabling OP-as-fuck RF mods to ignore your powertiers.
So while I appreciate the offer, I don't see myself bringing back EU compatibility, and while I do not object to your idea @malte0811 (a proper transforming station multiblock could be really cool!), I personally wouldn't be too comfortable using it, unless you find a way to balance it well.

**Comment by oitsjustjose** [comment]
Yup, as the last few months have progressed I've come to just live with the isolation that IC2's energy net has always brought - thank you guys for the considerations and discussion, but I side with Blu here, FTB:IE basically turned all of Immersive Engineering into "that conversion mod" and didn't focus too heavily on the actual content that Immersive Engineering brings. Likewise, it allowed people to bypass some of the more challenging (or, in my opinion, entertaining) aspects of IC2: proper power management. 

Like I said - thanks all - gonna close this issue because I don't expect anything to be implemented, and I'd say that's completely reasonable.

**Comment by malte0811** [comment]
The addon is done now, I am just waiting on the next IE release which has a little change that allows me to use the `BlockIEBase` class for the addon (fe885f7307069b1aacf65c994455ddf8cd628674). As you see in the screenshot below, transfer from or to cables works fine (the machines on the right are all running).

![2016-09-14_18 24 51](https://cloud.githubusercontent.com/assets/10406104/18520694/39930130-7aa9-11e6-80bf-362642ba79b8.png)

@Aroma1997 There is one other thing I need though: reliable values for the loss of copper and gold cables. The wiki states that the loss rate is 0.025 EU/block for tin cable and 0.8 EU/block for (insulated) HV cable. Are those values correct?

**Comment by mindforger** [comment]
you forogt the glass fiber cable, their loss is waaaay less

**Comment by Aroma1997** [comment]
Dunno the cable loss out of my head. It's schown in the cable tooltip though.

**Comment by malte0811** [comment]
IndustrialWires is available [here](https://minecraft.curseforge.com/projects/industrial-wires) now.
@Aroma1997 Thanks, I didn't notice that tooltip for some reason.
@mindforger I haven't added glass fiber wires yet, I need to find a way to make them properly balanced. The other wires are  cheaper per block than the IC2 cables which isn't too much of a problem for "normal" wires since those are pretty cheap. I have an idea on making the wires more balanced, but I will have to see how doable it is with IE and Forge.

**Comment by Degrath** [comment]
thanks! that makes everything clear and also points to an addon I didn't know existed for the project. Very helpful
